rsync-downloaded-files.sh

```diff
@@ -1,21 +1,41 @@
 #!/bin/bash
 
+set -e
+
 # Iterate over all the collections that are fully loaded into the database
 
 for collection in $(psql -U ocdskfpreadonly -d ocdskingfisherprocess -h localhost -c "select source_id, to_char(data_version, 'YYYYMMDD_HH24MISS') from collection where store_end_at is not null and transform_type is null" -qAtX | tr "|" "/"); do
 
+        echo "COLLECTION ${collection}"
+
         scraper=$(echo ${collection} | awk 'BEGIN { FS = "/" } { print $1 }')
         timestamp=$(echo ${collection} | awk 'BEGIN { FS = "/" } { print $2 }')
 
-        # Rsync to the disk server
-	ssh -o StrictHostKeyChecking=no archive@archive.kingfisher.open-contracting.org "mkdir -p /home/archive/data/${collection}"
 
-        rsync -r -e "ssh -o StrictHostKeyChecking=no" /home/ocdskfs/scrapyd/data/${collection}/ archive@archive.kingfisher.open-contracting.org:/home/archive/data/${collection}/ || continue
+        # Find folder on source server
+	# LOCAL_DIR serves 2 purposes; the data may be in one of 2 places AND more importantly, if the data has already been successfully archived, nothing is done.
+	LOCAL_DIR=""
+	if [ -d "/home/ocdskfs/scrapyd/data/${collection}/" ]; then 
+		LOCAL_DIR="/home/ocdskfs/scrapyd/data/${collection}/"
+	elif [ -d "/home/ocdskfs/old-data/${collection}" ]; then 
+		LOCAL_DIR="/home/ocdskfs/old-data/${collection}/"
+	fi
+	
+        if [ "${LOCAL_DIR}" ]; then
+
+		# Create folder on disk server
+		ssh -o StrictHostKeyChecking=no archive@archive.kingfisher.open-contracting.org "mkdir -p /home/archive/data/${collection}"
```

Comment:
Not a blocker now, but the `StrictHostKeyChecking=no` was just to get around chicken/egg with Salt development. Keys and auth_keys really should be managed by salt.

Reply:
Noted, thanks